pkg/controllers/cloud/networkpolicy_controller.go

```diff
@@ -318,95 +314,46 @@ func (r *NetworkPolicyReconciler) processMemberGrp(name string, eventType watch.
 				delete(removedMembers, vpc)
 			}
 			if err := sg.update(members, removed, r); err != nil {
-				r.Log.Error(err, "Update SecurityGroup to cloud", "key", key)
+				r.Log.Error(err, "update SecurityGroup to cloud", "key", key)
 				continue
 			}
 		} else {
-			// Find accountID and cloudProvider for the cloud resource using any member inside as fields are same for all
-			for _, a := range members {
-				cloudProvider = a.CloudProvider
-				accountID = a.AccountID
-				break
-			}
+			// All members in a vpc will share same AccountID and CloudProvider.
 			cloudRsrc := securitygroup.CloudResource{
 				Type:            securitygroup.CloudResourceTypeVM,
 				CloudResourceID: *key,
-				AccountID:       accountID,
-				CloudProvider:   cloudProvider,
+				AccountID:       members[0].AccountID,
+				CloudProvider:   members[0].CloudProvider,
 			}
-			sg = creator(&cloudRsrc, members, nil)
+			sg = creatorFunc(&cloudRsrc, members, nil)
 			if sg == nil {
+				r.Log.Error(fmt.Errorf("failed to create"), "cloud resource", "resource", cloudRsrc)
```

Comment:
here its creating a addrSecurityGroup{} struct.
So the log message is misleading.